examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java

```diff
@@ -0,0 +1,128 @@
+
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  private static SslContext buildSslContext(String trustCertCollectionFilePath,
+                                            String clientCertChainFilePath,
+                                            String clientPrivateKeyFilePath) throws SSLException {
+    SslContextBuilder builder = GrpcSslContexts.forClient();
+    if (trustCertCollectionFilePath != null) {
+      builder.trustManager(new File(trustCertCollectionFilePath));
+    }
+    if (clientCertChainFilePath != null && clientPrivateKeyFilePath != null) {
+      builder.keyManager(new File(clientCertChainFilePath), new File(clientPrivateKeyFilePath));
+    }
+    return builder.build();
+  }
+
+  /**
+   * Construct client connecting to HelloWorld server at {@code host:port}.
+   */
+  public HelloWorldClientTls(String host,
+                             int port,
+                             SslContext sslContext) throws SSLException {
+
+    this(NettyChannelBuilder.forAddress(host, port)
+        .negotiationType(NegotiationType.TLS)
+        .sslContext(sslContext)
+        .build());
+  }
+
+  /**
+   * Construct client for accessing RouteGuide server using the existing channel.
+   */
+  HelloWorldClientTls(ManagedChannel channel) {
+    this.channel = channel;
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Say hello to server.
+   */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the first element of {@code args} is the name to use in the
+   * greeting.
+   */
+  public static void main(String[] args) throws Exception {
+
+    if (args.length < 2 || args.length == 4 || args.length > 5) {
+      System.out.println("USAGE: HelloWorldClientTls host port [trustCertCollectionFilePath] [clientCertChainFilePath] " +
+          "[clientPrivateKeyFilePath]\n  Note: clientCertChainFilePath and clientPrivateKeyFilePath are only needed " +
+          "if mutual auth is desired. And if you specify clientCertChainFilePath you must also specify " +
+          "clientPrivateKeyFilePath");
+      System.exit(0);
+    }
+
+    HelloWorldClientTls client = new HelloWorldClientTls(args[0],
+        Integer.parseInt(args[1]),
+        buildSslContext(args.length > 2 ? args[2] : null, args.length > 3 ? args[3] : null, args.length > 4 ? args[4] : null));
```

Comment:
I personally find this hard to read.   Maybe consider just switching on args.length, and constructing `client`  from the appropriate params?

Reply:
sure. will do